remote-api/src/test/java/org/alfresco/rest/api/model/rules/CompositeConditionTest.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * #%L
+ * Alfresco Remote API
+ * %%
+ * Copyright (C) 2005 - 2022 Alfresco Software Limited
+ * %%
+ * This file is part of the Alfresco software.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail.  Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+
+package org.alfresco.rest.api.model.rules;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.Serializable;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.alfresco.repo.action.ActionConditionImpl;
+import org.alfresco.repo.action.evaluator.ComparePropertyValueEvaluator;
+import org.alfresco.service.Experimental;
+import org.alfresco.service.cmr.action.ActionCondition;
+import org.assertj.core.api.Condition;
+import org.junit.Test;
+
+@Experimental
+public class CompositeConditionTest
+{
+
+    @Test
+    public void testFrom()
+    {
+        final List<ActionCondition> actionConditions = List.of(
+            createActionCondition(),
+            createActionCondition(true),
+            createActionCondition()
+        );
+
+        // when
+        final CompositeCondition compositeCondition = CompositeCondition.from(actionConditions);
+
+        assertThat(compositeCondition).is(havingCompositeConditions(2, false));
+        assertThat(compositeCondition.getCompositeConditions().get(0)).is(havingSimpleConditions(2, false));
+        assertThat(compositeCondition.getCompositeConditions().get(1)).is(havingSimpleConditions(1, true));
```

Comment:
This test is pretty hard to follow

Reply:
Simplified the test.

Reply:
Ah - I understand what the test is doing now.